### PR TITLE
Allow setting merchant for split bill

### PR DESCRIPTION
**Pull request by youssef-lr**
### Fixed Issues

$ https://github.com/Expensify/App/issues/29732
$ https://github.com/Expensify/App/issues/29691

### Tests

- [ ] Verify that no errors appear in the JS console

1. Create a manual split and make sure you set a merchant and a description.
2. Verify the split preview shows the merchant but not the description.
3. Create another split by supplying only the description, make sure the description is shown in the preview.

### QA Steps

- [x] Verify that no errors appear in the JS console

1. Create a manual split and make sure you set a merchant and a description.
2. Verify the split preview shows the merchant but not the description.
3. Create another split by supplying only the description, make sure the description is shown in the preview.

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by youssef-lr** [comment]
Done @aimane-chnaif

**Comment by youssef-lr** [comment]
bump @aimane-chnaif

File: src/components/ReportActionItem/MoneyRequestPreview.js
Comment (youssef-lr) [inline src/components/ReportActionItem/MoneyRequestPreview.js:172]: Our design is we either show the merchant if it's not the default one or empty, otherwise we show the description.

Comment (youssef-lr) [reply src/components/ReportActionItem/MoneyRequestPreview.js:172]: cc @trjExpensify @shawnborton this is going to fix the merchant having a small font size as well

Screenshot 2023-11-14 at 22 17 09

Comment (trjExpensify) [reply src/components/ReportActionItem/MoneyRequestPreview.js:172]: That looks good to me, but I'll defer to Shawn.

Comment (shawnborton) [reply src/components/ReportActionItem/MoneyRequestPreview.js:172]: That looks good to me. Can you also take a look at [this one](https://expensify.slack.com/archives/C049HHMV9SM/p1700000146978699) while you are here? Thanks!

**Comment by marcochavezf** [comment]
Hmm if I understand correctly, this PR and https://github.com/Expensify/App/pull/31604 are fixing the problem, correct?

**Comment by marcochavezf** [comment]
Ok I see, in that case yeah I think it makes sense to hold this PR for https://github.com/Expensify/App/pull/31604 cc @trjExpensify 

I've updated this PR to get up to date with the main branch.

**Comment by marcochavezf** [comment]
@aimane-chnaif thoughts about proceeding with the review of this PR, given the other one was reverted?

**Comment by youssef-lr** [comment]
bump @aimane-chnaif, PR is updated

**Comment by youssef-lr** [comment]
Done @aimane-chnaif

**Comment by youssef-lr** [comment]
oh it seems like I need to copy the changes to `MoneyTemporaryForRefactorRequestConfirmationList`

**Comment by youssef-lr** [comment]
It looks like it's still being used now? Are you thinking of `MoneyRequestConfirmationList`?

**Comment by youssef-lr** [comment]
Ah, I was looking for one with that has `Temporary` in the name, thanks! Will update.

**Comment by youssef-lr** [comment]
@aimane-chnaif I pushed a slighty unrelated fix to this PR, it has to do with a bug where we show the merchant & date in scanned requests when they should be hidden.

**Comment by youssef-lr** [comment]
I think this PR fixes that issue? Why not just merge this?

File: src/pages/iou/request/step/IOURequestStepConfirmation.js
Comment (youssef-lr) [reply src/pages/iou/request/step/IOURequestStepConfirmation.js:220]: Updated

Comment (youssef-lr) [reply src/pages/iou/request/step/IOURequestStepConfirmation.js:220]: Sounds good

**Comment by youssef-lr** [comment]
Updated @aimane-chnaif, please double check if I missed something

File: src/pages/iou/request/step/IOURequestStepConfirmation.js
Comment (tgolen) [reply src/pages/iou/request/step/IOURequestStepConfirmation.js:220]: Let's do the fix separately, or ideally, you can add the fix to both.

Comment (youssef-lr) [reply src/pages/iou/request/step/IOURequestStepConfirmation.js:220]: >To make everything consistent, let's define order something like this:

I reverted the the re-ordering of the params, I think let's keep this PR light and merge it as soon as we can as it's been dragging for too long, we can clean up the ordering of params later. Sounds good?

File: src/libs/actions/IOU.js
Comment (youssef-lr) [reply src/libs/actions/IOU.js:1033]: No I don't think this was part of the commit that reordered params

Comment (youssef-lr) [reply src/libs/actions/IOU.js:1033]: I just reordered the merchant here, not the all of the params

File: src/components/MoneyTemporaryForRefactorRequestConfirmationList.js
Comment (tgolen) [inline src/components/MoneyTemporaryForRefactorRequestConfirmationList.js:144]: This prop is still being passed to the component.

File: src/components/ReportActionItem/MoneyRequestPreview.js
Comment (tgolen) [inline src/components/ReportActionItem/MoneyRequestPreview.js:179]: I suggest starting off this variable as `requestMerchant` rather than an empty string. That allows the removal of the `else` below.

File: src/libs/actions/IOU.js
Comment (tgolen) [reply src/libs/actions/IOU.js:1033]: This has been touched quite a few times lately, so just be sure to merge `main` and then ensure everything is still passed in the proper order before merging this.

**Review by tgolen — CHANGES_REQUESTED** [review]

File: src/components/ReportActionItem/MoneyRequestPreview.js
Comment (youssef-lr) [reply src/components/ReportActionItem/MoneyRequestPreview.js:177]: Agreed!

File: src/libs/actions/IOU.js
Comment (youssef-lr) [reply src/libs/actions/IOU.js:1071]: I just merged main and fixed conflicts. Thanks!

**Comment by youssef-lr** [comment]
Updated @tgolen @aimane-chnaif

File: src/components/ReportActionItem/MoneyRequestPreview.js
Comment (tgolen) [reply src/components/ReportActionItem/MoneyRequestPreview.js:177]: It doesn't look like this was included in the recent changes?

**Review by tgolen — CHANGES_REQUESTED** [review]

File: src/components/ReportActionItem/MoneyRequestPreview.js
Comment (youssef-lr) [reply src/components/ReportActionItem/MoneyRequestPreview.js]: Sorry the push didn't go through, it's updated now.

**Review by tgolen — CHANGES_REQUESTED** [review]
Cool, that looks better! Looks like the linter found one more `isScanRequest` that can be removed.

**Comment by youssef-lr** [comment]
Fixed, thanks!

**Comment by marcochavezf** [comment]
Fixed merged conflict

**Comment by youssef-lr** [comment]
bump for review please @aimane-chnaif

**Review by tgolen — DISMISSED** [review]

**Review by aimane-chnaif — CHANGES_REQUESTED** [review]

File: src/pages/iou/request/step/IOURequestStepConfirmation.js
Comment (youssef-lr) [reply src/pages/iou/request/step/IOURequestStepConfirmation.js:224]: Done

**Comment by youssef-lr** [comment]
bump @aimane-chnaif, can we try and get this merged today please

File: src/components/ReportActionItem/MoneyRequestPreview.js
Comment (youssef-lr) [reply src/components/ReportActionItem/MoneyRequestPreview.js]: Updated, I think it doesn't look right

**Comment by youssef-lr** [comment]
Sorry I missed that, that's an intentional change in this PR. SmartScan should take care of those fields and not the user.

**Review by tgolen — APPROVED** [review]

**Comment by youssef-lr** [comment]
bump @aimane-chnaif

**Comment by youssef-lr** [comment]
Yeah it's out of scope. I'll look into that though. Thanks for catching this!

File: src/libs/actions/IOU.js
Comment (youssef-lr) [reply src/libs/actions/IOU.js:1159]: It's okay, we're going to deprecate the `Request` merchant soon on P2P requests. Here's the plan from a private issue:

> Make manual IOU expenses only have manually entered amount and description:
> a. Don't show Merchant or Date fields in a hidden "detailed" section
> b. Don't show Merchant: Request this is confusing and seems broken
>
> Show Merchant and/or Date on SmartScanned IOU expenses only if they succeeded.
> a. Don't show Merchant: Unknown
> b. Don't show a RBR if the merchant isn't read
> c. If the SmartScan fails to get one or the other, just hide that field -- exactly like a manual request
>
> For expenses on expense reports, however, require Merchant, Date, and Amount -- and show a RBR if any are missing, for both manual and SmartScanned expenses

**Review by aimane-chnaif — APPROVED** [review]

File: src/pages/iou/request/step/IOURequestStepConfirmation.js
Comment (getusha) [inline src/pages/iou/request/step/IOURequestStepConfirmation.js:340]: this change caused a regression https://github.com/Expensify/App/issues/33588

**Pull request merged**
